Realm/Realm/Handles/ObjectHandle.cs

```diff
@@ -115,26 +113,9 @@ public ObjectHandle(RealmHandle root, IntPtr handle) : base(root, handle)
         {
         }
 
-        public override bool Equals(object obj)
+        public bool ObjEquals(ObjectHandle other)
```

Comment:
This was replaced as it wasn't correct to override `ObjectHandle.Equals` as we're not really checking for equality of the ObjectHandle struct, but rather the value that it represents (e.g. two different ObjectHandle instances pointing to the same underlying object are not necessarily "equal" as they may have different unbind lists and parents).